pkg/secrethub/service_aws.go

```diff
@@ -0,0 +1,35 @@
+package secrethub
+
+import (
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/aws"
+
+	awssdk "github.com/aws/aws-sdk-go/aws"
+)
+
+// ServiceAWSService handles operations on service accounts from SecretHub.
+type ServiceAWSService interface {
+	// Create creates a new service account for the given repo.
+	Create(path string, description string, keyID, role string, cfgs ...*awssdk.Config) (*api.Service, error)
+}
+
+func newServiceAWSService(client client, s ServiceService) ServiceAWSService {
+	return serviceAWSService{
+		client:         client,
+		serviceService: s,
+	}
+}
+
+type serviceAWSService struct {
+	client         client
+	serviceService ServiceService
+}
+
+// Create creates a new AWS service account for the given repo.
+func (s serviceAWSService) Create(path string, description string, keyID, role string, cfgs ...*awssdk.Config) (*api.Service, error) {
+	creator, err := aws.NewServiceCreator(keyID, role, cfgs...)
+	if err != nil {
+		return nil, err
+	}
+	return s.serviceService.Create(path, description, creator, creator)
```

Comment:
Huh, interesting... You're calling the `Create` again here. I'm starting to wonder whether the `create` with separate `verifier` and `encrypter` arguments should be private and we expose only public functions that are specific to the type of credential?